de.dlr.sc.virsat.project.ui/src/de/dlr/sc/virsat/project/ui/navigator/dropAssist/DVLMObjectDropAdapterAssistant.java

```diff
@@ -9,98 +9,176 @@
  *******************************************************************************/
 package de.dlr.sc.virsat.project.ui.navigator.dropAssist;
 
-import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
 
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.Status;
-import org.eclipse.emf.common.command.Command;
-import org.eclipse.emf.ecore.EObject;
-import org.eclipse.jface.util.LocalSelectionTransfer;
-import org.eclipse.jface.viewers.ISelection;
 import org.eclipse.jface.viewers.IStructuredSelection;
-import org.eclipse.jface.viewers.StructuredSelection;
-import org.eclipse.swt.dnd.DND;
 import org.eclipse.swt.dnd.DropTargetEvent;
 import org.eclipse.swt.dnd.TransferData;
-import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.navigator.CommonDropAdapter;
 import org.eclipse.ui.navigator.CommonDropAdapterAssistant;
-import org.eclipse.ui.navigator.CommonViewer;
-
-import de.dlr.sc.virsat.project.editingDomain.VirSatEditingDomainRegistry;
-import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
-import de.dlr.sc.virsat.project.editingDomain.commands.VirSatEditingDomainClipBoard.ClipboardState;
-import de.dlr.sc.virsat.project.editingDomain.commands.VirSatPasteFromClipboardCommand;
-import de.dlr.sc.virsat.project.ui.navigator.VirSatNavigator;
+import de.dlr.sc.virsat.model.dvlm.categories.ATypeInstance;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.concepts.util.ActiveConceptHelper;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.project.Activator;
 
 /**
- * VirSat Drop Assitant to copy and paste SEIs and CAs in the Navigator
- * @author fisc_ph
- *
+ * Drop Assistant for VirSat Navigator that delegates calls to concept specific drop assistants
  */
 public class DVLMObjectDropAdapterAssistant extends CommonDropAdapterAssistant {

```

Comment:
Good question! @SaMuellerDLR  Do we need the drop adapter there? I think it is heritage and we should leave it there for the moment. But i guess we don't need it , since we do not have any code dealing with Copy&Paste and DnD of Equations. Do we?

Reply:
I checked it yesterday and we dont have any drag and drop functionality for equations.
I found the commit in our old repository, and there it looks like the entry only exists due to creating the setup with copy & paste. I think it can be safely removed.

Reply:
I removed it from calculations